geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/streams/GeoMesaStreamsBuilderTest.scala

```diff
@@ -0,0 +1,235 @@
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.{ByteArrayDeserializer, LongDeserializer, StringDeserializer}
+import org.apache.kafka.streams
+import org.apache.kafka.streams.kstream.{Transformer, TransformerSupplier}
+import org.apache.kafka.streams.processor.{ProcessorContext, WallclockTimestampExtractor}
+import org.apache.kafka.streams.{StreamsConfig, TopologyTestDriver}
+import org.geotools.data.{DataStoreFinder, Query, Transaction}
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.kafka.EmbeddedKafka
+import org.locationtech.geomesa.kafka.data.KafkaDataStore
+import org.locationtech.geomesa.utils.collection.SelfClosingIterator
+import org.locationtech.geomesa.utils.geotools.converters.FastConverter
+import org.locationtech.geomesa.utils.geotools.{FeatureUtils, SimpleFeatureTypes}
+import org.locationtech.geomesa.utils.io.WithClose
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+import java.nio.charset.StandardCharsets
+import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
+import java.util.{Collections, Date, Properties}
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration.DurationInt
+
+@RunWith(classOf[JUnitRunner])
+class GeoMesaStreamsBuilderTest extends Specification with StrictLogging {
+
+  import org.apache.kafka.streams.scala.ImplicitConversions._
+  import org.apache.kafka.streams.scala.Serdes._
+  import org.apache.kafka.streams.scala.kstream._
+
+  import scala.collection.JavaConverters._
+
+  lazy val sft =
+    SimpleFeatureTypes.createImmutableType("streams", "name:String,age:Int,dtg:Date,*geom:Point:srid=4326")
+
+  lazy val features = Seq.tabulate(10) { i =>
+    ScalaSimpleFeature.create(sft, s"id$i", s"name$i", i % 2, s"2022-04-27T00:00:0$i.00Z", s"POINT(1 $i)")
+  }
+
+  var kafka: EmbeddedKafka = _
+
+  step {
+    logger.info("Starting embedded kafka/zk")
+    kafka = new EmbeddedKafka()
+    logger.info("Started embedded kafka/zk")
+  }
+
+  private val zkPaths = Collections.newSetFromMap(new ConcurrentHashMap[String, java.lang.Boolean]())
+
+  def getParams(zkPath: String): Map[String, String] = {
+    require(zkPaths.add(zkPath), s"zk path '$zkPath' is reused between tests, may cause conflicts")
+    Map(
+      "kafka.brokers"            -> kafka.brokers,
+      "kafka.zookeepers"         -> kafka.zookeepers,
+      "kafka.topic.partitions"   -> "1",
+      "kafka.topic.replication"  -> "1",
+      "kafka.consumer.read-back" -> "Inf",
+      "kafka.zk.path"            -> zkPath
+    )
+  }
+
+  "GeoMesaStreamsBuilder" should {
+    "read from geomesa topics" in {
+      val params = getParams("word/count")
+      // write features to the embedded kafka
+      val kryoTopic = WithClose(DataStoreFinder.getDataStore(params.asJava)) { ds =>
```

Comment:
Will the streams integration work with Avro topics?

Reply:
yep!